pkg/cmd/reset.go

```diff
@@ -100,5 +105,41 @@ func runReset(opts *resetOpts) error {
 
 	s.Logger.Warnln("this command will require an explicit confirmation starting with the next minor release (v1.3)")
 
+	fmt.Println("The following nodes will be reset. The Kubernetes cluster running on those nodes will be permanently destroyed ")
+	for _, node := range s.Cluster.ControlPlane.Hosts {
+		fmt.Printf("\t+ reset control plane node %q (%s)\n", node.Hostname, node.PrivateAddress)
+	}
+	for _, node := range s.Cluster.StaticWorkers.Hosts {
+		fmt.Printf("\t+ reset static worker nodes %q (%s)\n", node.Hostname, node.PrivateAddress)
+	}
+
+	if s.DynamicClient == nil {
+		err = kubeconfig.BuildKubernetesClientset(s)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Gather information about machine-controller managed nodes
+	ctx := context.Background()
+	machines := v1alpha1.MachineList{}
+	if err := s.DynamicClient.List(ctx, &machines); err != nil {
```

Comment:
Let's use `s.Context` instead of `ctx`:

```suggestion
	if err := s.DynamicClient.List(s.Context, &machines); err != nil {
```